Feature/cv2-4502 articles trash action

sarahkeh opened the pull request:
- [x] Create button, add to edit slideout
- [x] Attach confirm proceed dialog, add copy from Brian
  - [x] Fact check: include link and title for media item
  - [x] Explainer: Only static message
- [x] Get data for attached media items, if the article has them (new data piece? existing data?)
- [x] Hook up Proceed to new trash API 
- [x] After user confirms trash: 
   - [x] Slideout is closed
   - [x] List of articles is updated to no longer show article
   - [x] Media items that were linked to the article no longer show article
   - [x] Number of articles goes down
## Description

Added a Move to Trash button that toggles the new trashed setting for articles.  If the article is active, when the user selects Edit, they have an option to move the article to the trash.  This will trigger a reload of the page at the moment, with proper Relay updates to come later. If the article is marked as trashed, they will now see a Restore from Trash button with a new Undo icon.  

## Type of change

- [ ] Performance improvement and/or refactoring (non-breaking change that keeps existing functionality)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Security mitigation or enhancement
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Automated test (add or update automated tests)

## How has this been tested?

Manual testing:
- verified button only appears in Edit mode
- Trashed Fact Check with Media item and verified article is no longer associated, cannot find Article in search
- Trashed Explainer linked to Media Item and verified article is no longer associated, cannot find Article in search
- Trashed Explainer without linked Media, no longer appears in list or search
- Trashed Fact Check without media item, no longer appears in list or search
- After refresh, article count decreases by one
- After refresh, the list of Explainers/Fact-Checks no longer contains that article.  

## Things to pay attention to during code review

Restore from Trash and Move to Trash are the same button but with different settings. Right now, without a list of Trashed items, the Restore from Trash can't quite be tested. This portion was also listed as a part of the List feature, so the testing of that side can be saved until the list is in place and the trashed articles can be viewed.  The page refresh is a temporary solution until the Relay changes are made, but this will be handled in a separate ticket.  

## Checklist

- [X] I have performed a self-review of my own code
- [X] I've made sure my branch is runnable and given good testing steps in the PR description
- [X] I considered secure coding practices when writing this code. Any security concerns are noted above.
- [X] I have commented my code in hard-to-understand areas, if any
- [ ] I have made needed changes to the README
- [X] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [X] If I implemented any new components, they are self-contained, their `propTypes` are declared and they use React Hooks and, if data-fetching is required, they use Relay Modern with fragment containers
- [X] If my components involve user interaction - specifically button, text fields, or other inputs - I have added a [BEM-like class name](https://meedan.atlassian.net/wiki/spaces/ENG/pages/1327628289/Naming+conventions+for+interactive+elements) to the element that is interacted with
- [ ] To the best of my knowledge, any new styles are applied according to the design system
- [ ] If I added a new external dependency, I included a rationale for doing so and an estimate of the change in bundle size (e.g., checked in https://bundlephobia.com/)
- [X] If I touched a file that included an eslint-disable-file header, I updated the code such that the disabler can be removed

brianfleming reviewed (changes requested):
Looking great @sarahkeh couple very nit picky things:

1. For the main text of the confirm dialogs I think you should add tagName="p" to the FormattedMessage component so the layout give a little space between the paragraph and the <Alert... component. See the screenshots below vs the similar delete list dialog:

<img width="611" alt="Screenshot 2024-09-19 at 10 52 33 AM" src="https://github.com/user-attachments/assets/9ea8b2c8-05cd-4902-9f07-10a305fb86d9">

## Versus
<img width="611" alt="Screenshot 2024-09-19 at 10 54 12 AM" src="https://github.com/user-attachments/assets/6bfda9d8-f761-45d1-a5a6-6e2fad23258c">


2. Seeing the dialogs in real life makes me think that ^^ main text is visually too long and should be separated out to make it more readable. Can you change it so it ends of being separate paragraphs? Same text really, but like this:

### Fact-Checks:

> Are you sure? If you send this article to the trash it will no longer be available to be associated with media responses.
>
> It can be removed from the Trash later, but it will be deleted permanently after 30 days.

### Explainers:

> Are you sure? If you send this article to the trash it will no longer be available to be associated with media responses.
> 
> If this article is currently associated with media, the link between the media and article will be removed.
> 
> It can be linked again if the article is removed from the Trash. After 30 days, this article will be permanently deleted.

brianfleming commented on src/app/components/article/ArticleTrash.js:
```diff
@@ -0,0 +1,258 @@
+
+  const factCheckBody = type === 'fact-check' && article.claim_description?.project_media ? (
+    <>
+      <FormattedMessage
```
There are others, just wanted to point out one of them

brianfleming commented:
@sarahkeh one other thing i forgot to mention in my review and not sure if this is a local data thing or a bug, but take a look at this screenshot. Its me sending a fact-check to the trash, the message doesn't include the alert, but mentions that its associated with media, which it isn't. So I am just unsure of why I am see this dialog:

<img width="1093" alt="Screenshot 2024-09-19 at 11 06 25 AM" src="https://github.com/user-attachments/assets/5a3f705e-30af-4bf5-8e43-eeebcef8ffd5">

sarahkeh commented:
@brianfleming I think I got everything:

1. added p tags 
2. Broke up messages (this is probably better for translation, too, so repetitive sentences aren't translated more than once" 
3. fixed that Claim/fact-check without media issue

<img width="675" alt="Screenshot 2024-09-20 at 10 10 55 AM" src="https://github.com/user-attachments/assets/e767f26a-aab6-48f5-9749-8b96872742a3">
<img width="691" alt="Screenshot 2024-09-20 at 10 12 06 AM" src="https://github.com/user-attachments/assets/0aabfee1-4445-4d61-9a76-e57398c01d90">
<img width="732" alt="Screenshot 2024-09-20 at 10 11 53 AM" src="https://github.com/user-attachments/assets/351b65c4-de10-4226-ab63-90ea1abcc31b">

brianfleming reviewed (approved):
nice! thanks for the changes @sarahkeh approving, but since I only play a developer on TV, make sure one of the real ones takes a look!

![trash-panda](https://github.com/user-attachments/assets/967961d0-ab45-44b6-9fe8-74ce647e4114)

brianfleming commented:
Oh, one thing @sarahkeh 
since we don't have restore yet, or the list yet, maybe this and all those things should go into an epic branch instead of develop?
cc @caiosba @amoedoamorim

caiosba commented:
Good point, @brianfleming ! @sarahkeh please create an epic branch `epic/CV2-4441-articles-trash` and make it the base branch for this PR. We can merge the epic branch to `develop` once CV2-5069 and CV2-5319 are done.

amoedoamorim reviewed (approved):
:lgtm:
